stan/math/gpu/opencl_context.hpp

```diff
@@ -73,13 +73,21 @@ class opencl_context_base {
     try {
       // platform
       cl::Platform::get(&platforms_);
+      if (OPENCL_PLATFORM_ID >= platforms_.size()) {
+        system_error("OpenCL Initialization", "[Platform]", -1,
+                       "CL_INVALID_PLATFORM");
+      }
```

Comment:
If a user specified an ID that did not exist, the test produced a seg fault. We now output the following:
```
[----------] Global test environment set-up.
[----------] 3 tests from MathGpu
[ RUN      ] MathGpu.getInfo
unknown file: Failure
C++ exception with description "OpenCL Initialization: [Platform] CL_INVALID_PLATFORM: Unknown error -1" thrown in the test body.
[  FAILED  ] MathGpu.getInfo (29 ms)
```